pymc/gp/hsgp_approx.py

```
@@ -85,6 +88,67 @@ def calc_basis_periodic(
+
+    In practice, you need to choose `c` large enough to handle the largest lengthscales,
+    and `m` large enough to accommodate the smallest lengthscales.
+
+    NB: These recommendations are based on a one-dimensional GP.
+
+    Parameters
+    ----------
+    x_range : List[float]
+        The range of the input variable on which the GP is going to be evaluated.
+        Should be a list with two elements [x_min, x_max].
+    lengthscale_range : List[float]
+        The range of the lengthscales. Should be a list with two elements [lengthscale_min, lengthscale_max].
+    cov_func : str
+        The covariance function to use. Supported options are "expquad", "matern52", and "matern32".
+
+    Returns
+    -------
+    Tuple[int, float, float]
+        A tuple containing the recommended values for `m`, `c`, and `S`.
+        - `m` : int
+            Number of basis vectors. Increasing it helps approximate smaller lengthscales, but increases computational cost.
+        - `c` : float
+            Scaling factor such that L = c * S, where L is the boundary of the approximation.
+            Increasing it helps approximate larger lengthscales, but may require increasing m.
+        - `S` : float
+            The value of `S`, which is half the range of `x`.
+
+    Raises
+    ------
+    ValueError
+        If either `x_range` or `lengthscale_range` is not in the correct order.
+
+    References
+    ----------
+    - Ruitort-Mayol, G., Anderson, M., Solin, A., Vehtari, A. (2022).
+    Practical Hilbert Space Approximate Bayesian Gaussian Processes for Probabilistic Programming
+    """
+    if (x_range[0] >= x_range[1]) or (lengthscale_range[0] >= lengthscale_range[1]):
+        raise ValueError("One of the boundaries out of order")
+
+    S = (x_range[1] - x_range[0]) / 2
+
+    if cov_func.lower() == "expquad":
+        a1, a2 = 3.2, 1.75
+
+    elif cov_func.lower() == "matern52":
+        a1, a2 = 4.1, 2.65
+
+    elif cov_func.lower() == "matern32":
```

Shall we raise an error with a friendly error message if `cov_function` does not match the expected ones?

---

Good point. Done